dlt/destinations/impl/filesystem/sql_client.py

```diff
@@ -0,0 +1,94 @@
+from typing import Any, Iterator, AnyStr, List
+
+import os
+
+import duckdb
+
+import sqlglot
+import sqlglot.expressions as exp
+
+from contextlib import contextmanager
+
+from dlt.common.destination.reference import DBApiCursor
+
+from dlt.destinations.sql_client import raise_database_error
+from dlt.destinations.fs_client import FSClientBase
+
+from dlt.destinations.impl.duckdb.sql_client import DuckDbSqlClient
+from dlt.destinations.impl.duckdb.factory import duckdb as duckdb_factory
+
+
+class FilesystemSqlClient(DuckDbSqlClient):
+    def __init__(self, fs_client: FSClientBase, protocol: str) -> None:
+        """For now we do all operations in the memory dataset"""
+        """TODO: is this ok?"""
+        super().__init__(
+            dataset_name="memory",
```

Comment:
we need exactly the same dataset name that was used in the pipeline and was passed to JobClient

Reply:
this is done now!